lib/display.py

```diff
@@ -136,6 +138,25 @@ def quit(data):
     prep_data_on_close(data)
 
 
+class actionCompleter(object):
+
+    def __init__(self):
+        return
+
+    def set_actions(self, actions):
+        self.actions = sorted(actions)
+
+    def complete(self, action, index):
+        if index == 0:
```

Comment:
`complete` is called multiple times per input `action` if the previous call to it returned a string, in this case `index` is incremented, if `None` is returned it assumes all possible completions have been returned and goes on its way. We only build the match list the first iteration to save a bit of time.